src/relion/zocalo/wrapper.py

```diff
@@ -145,6 +147,41 @@ def run(self):
                 )
                 logger.info("Sent %d commands to ISPyB", len(ispyb_command_list))
 
+            ### Extract and send Icebreaker results as histograms if the Icebreaker grouping job has run
+            if not self.opts.stop_after_ctf_estimation:
+                icebreaker_file_path = (
+                    self.working_directory / "External/Icebreaker_group/"
+                )
+                attachment_list = []
+                try:
+                    pdf_file_name = icebreaker_histogram.create_pdf_histogram(
+                        self.working_directory
+                    )
+                    json_file_name = icebreaker_histogram.create_json_histogram(
+                        self.working_directory
+                    )
+                    if pdf_file_name is None or json_file_name is None:
+                        pass
+                    elif should_send_icebreaker:
+                        attachment_list.append(
+                            ispyb_attachment(
+                                json_file_name, str(icebreaker_file_path), "Graph"
```

Comment:
so if `json_file_name` (or rather, the return value of `create_json_histogram()`) was a path object, then you can here just call `ispyb_attachment(value.parent, value.path, "Graph")` or even rewrite `ispyb_attachment()` to take a path object directly. That means you won't need to create `icebreaker_file_path` at all in this function.